File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        left, right = 0, 0

        #while (r > 0 and numbers[r] > target):
        #    r -= 1
        while l < r:
            if numbers[l] + numbers[r] > target:
                r -= 1
            elif numbers[l] + numbers[r] < target:
                l += 1
            else:
                left = l + 1
                right = r + 1
                break
        return [left, right]


        

File: Data Structures & Algorithms/binary-search/submission-0.java
public class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {
            int temp = left + ((right - left) / 2);
            if (nums[temp] > target) {
                right = temp - 1;
            } else if (nums[temp] < target) {
                left = temp + 1;
            } else {
                return temp;
            }
        }
        return -1;
    }
}